### PR TITLE
Picker ItemsSource fixes

### LINKED ISSUE #19702: Picker goes into infinite loop when updating ItemsSource ObservableCollection during SelectedIndex update
### Description

When updating an ObservableCollection bound to the ItemsSource of a Picker in any way other than Add or Remove in an event handler for a change in SelectedIndex, the Picker gets stuck in an infinite loop bouncing back and forth between -1 and the new SelectedIndex property. May also be related to #8874. 

### Steps to Reproduce

Execute the following code:
```
var items = new ObservableCollection<object>
{
	new { Name = "John" },
	"Paul",
	"Ringo"
};
var picker = new Picker
{
	ItemDisplayBinding = new Binding("Name"),
	ItemsSource = items,
};

picker.PropertyChanged += (sender, e) =>
{
	if (e.PropertyName == nameof(Picker.SelectedIndex))
	{
		items.Clear();
	}
};

picker.SelectedIndex = 1;
```

Expected outcome: The picker Items collection is empty and SeletedIndex gets set to -1.
Actual outcome: The code never finishes, and is stuck in an infinite loop.

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.3

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android, Windows

### Affected platform versions

Windows 11, Android 11

### Did you find any workaround?

Know working workaround is to modify the ObservableCollection using only Add, Insert, and Remove operations. Another possible workaround would be to work with the Items collection directly.

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by BurningLights**
### Description of Change

Fixes the Picker getting stuck in an infinite loop when the ItemsSource is assigned as an ObservableCollection and the collection is modified during a change handler when the SelectedIndex changes.

### Issues Fixed

Fixes #19702

### CODE REVIEW

**Comment by rmarinho** [comment]
Can you rebase on main please?

**Comment by BurningLights** [comment]
@dotnet-policy-service agree

**Comment by SarthakB26** [comment]
I am experiencing this too. Picker behave very weirdly when inside collection view.

File: src/Controls/tests/Core.UnitTests/ScrollViewUnitTests.cs [hunk 1]
```diff
@@ -267,12 +267,12 @@ public void TestScrollToElementNotAnimated()
 		}
 
 		[Fact]
-		public void TestScrollToInvalid()
+		public async void TestScrollToInvalid()
```
Comment (jsuarezruiz) [inline src/Controls/tests/Core.UnitTests/ScrollViewUnitTests.cs:270]: `async Task`

**Review by PureWeen — APPROVED** [review]

**Comment by kubaflo** [comment]
Looks fine to me! Maybe a UI test would be useful, but I can add it in this PR: https://github.com/dotnet/maui/pull/22353

**Pull request merged**